### PR TITLE
Change owner address from SC cross shard

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- There was an issue with the `ChangeOwnerAddress` built-in function that needed to be fixed. The problem occurred when a smart contract was the owner of another smart contract and those smart contracts were in different shards. In such cases, the contract that had the ownership could not set a new owner for the second contract because the `ChangeOwnerAddress` built-in function would not work

- Changed the version of the `mx-chain-vm-common-go` module
## Proposed changes
- Added a semi-integration test for the case above
- PR from mx-chain-vm-common: https://github.com/multiversx/mx-chain-vm-common-go/pull/243
- PR from mx-chain-vm-go: https://github.com/multiversx/mx-chain-vm-go/pull/790

## Testing procedure
Steps:
1. Deploy a contract on the `shard0` 
2. Deploy a contract on `shard1` ( the test contract must have a function  that can call the built-in function `ChangeOwnerAddress` ) 
3. Set the owner of the first contract to be the second contract
4. Call the function `change_owner@firstContractAddress@newOwner` from the second contract and change the owner of the first contract.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -20,8 +20,8 @@ require (
 	github.com/multiversx/mx-chain-logger-go v1.0.13
 	github.com/multiversx/mx-chain-scenario-go v1.2.1
 	github.com/multiversx/mx-chain-storage-go v1.0.13
-	github.com/multiversx/mx-chain-vm-common-go v1.5.5
-	github.com/multiversx/mx-chain-vm-go v1.5.16
+	github.com/multiversx/mx-chain-vm-common-go v1.5.6-0.20231025104250-d24edf1e19ae
```
Comment (iulianpascalau) [inline go.mod:23]: let's not forget about proper release

File: common/enablers/epochFlags.go [hunk 2]
```diff
@@ -103,6 +103,7 @@ type epochFlagsHolder struct {
 	fixDelegationChangeOwnerOnAccountFlag       *atomic.Flag
 	dynamicGasCostForDataTrieStorageLoadFlag    *atomic.Flag
 	nftStopCreateFlag                           *atomic.Flag
+	changeOwnerAddressCrossShardThroughSCFlag   *atomic.Flag
```
Comment (iulianpascalau) [inline common/enablers/epochFlags.go:106]: missing tests in config/tomlConfig_test.go ?

**Review by sasurobert — APPROVED** [review]

File: common/enablers/epochFlags.go (hunk 2 above)
Comment (miiu96) [reply common/enablers/epochFlags.go:106]: done

File: go.mod (hunk 1 above)
Comment (miiu96) [reply go.mod:23]: I will add proper tags after system test

Comment (miiu96) [reply go.mod]: added

**Review by sasurobert — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: v1.5.13-dev-config-8c685c8e00 -> fixes-change-owner-from-sc-1b86927b7b

--- Specific errors ---

block hash does not match 10170
wrong nonce in block 3644
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  0
Nr. of all WARNS:  347
Nr. of new ERRORS:  0
Nr. of new WARNS:  20
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

--- WARNINGS ---

gcp-bel-validator-10 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  1

gcp-bel-validator-7 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  6

gcp-bel-validator-18 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  1

gcp-bel-validator-28 :
Warn: shard remained in an epoch that is too old shardID = shard Epoch = meta Epoch = err = missing header : GetMarshalizedHeaderFromStorage hash =  6
Warn: trie recreate error:           error = getNodeFromDB error: key not found for key root =  3

gcp-bel-validator-6 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  2

gcp-bel-validator-33 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  1

/------/

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**